plugins/pixart-rf/fu-pxi-wireless-device.c

```diff
@@ -0,0 +1,606 @@
+	FuPxiWirelessDevice *self = FU_PXI_WIRELESS_DEVICE (device);
+	guint8 buf[FU_PXI_RECEIVER_DEVICE_OTA_BUF_SZ] = { 0x0 };
+	guint16 checksum_device = 0x0;
+	guint8 status = 0x0;
+	g_autoptr(GByteArray) receiver_cmd = g_byte_array_new ();
+	g_autoptr(GByteArray) ota_cmd = g_byte_array_new ();
+
+	/* ota check crc command */
+	fu_byte_array_append_uint8 (ota_cmd, 0x3);					/* ota command length */
+	fu_byte_array_append_uint8 (ota_cmd, FU_PXI_DEVICE_CMD_FW_OTA_CHECK_CRC);	/* ota command */
+	fu_byte_array_append_uint16 (ota_cmd, checksum, G_LITTLE_ENDIAN);		/* checkesum */
+
+	/* increase the serial number */
+	self->sn++;
+
+	if (!fu_pxi_composite_receiver_cmd (FU_PXI_DEVICE_CMD_FW_OTA_CHECK_CRC,
+					    self->sn,
+					    self->model.target,
+					    receiver_cmd,
+					    ota_cmd,
+					    error))
+		return FALSE;
+	if (!fu_pxi_wireless_device_set_feature (FU_DEVICE (parent),
+						 receiver_cmd->data,
+						 receiver_cmd->len,
+						 error))
+		return FALSE;
+	if (!fu_pxi_wireless_device_get_cmd_response (self, buf, sizeof(buf), error))
+		return FALSE;
+	fu_common_dump_raw (G_LOG_DOMAIN, "crc buf", buf, sizeof(buf));
+	if (!fu_common_read_uint8_safe (buf, sizeof(buf), 0x5, &status, error))
+		return FALSE;
+	if (!fu_common_read_uint16_safe (buf, sizeof(buf), 0x6,
+					 &checksum_device, G_LITTLE_ENDIAN, error))
+		return FALSE;
+	if (status == OTA_RSP_CODE_ERROR) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_READ,
+			     "checksum fail, got 0x%04x, expected 0x%04x",
+			     checksum_device,
+			     checksum);
+		return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_pxi_wireless_device_fw_object_create (FuDevice *device, FuChunk *chk, GError **error)
+{
+	FuPxiWirelessDevice *self = FU_PXI_WIRELESS_DEVICE (device);
+	FuPxiReceiverDevice *parent = fu_pxi_wireless_device_get_parent (device ,error);
+	g_autoptr(GByteArray) receiver_cmd = g_byte_array_new ();
+	g_autoptr(GByteArray) ota_cmd = g_byte_array_new ();
+
+	/* ota object create command */
+	fu_byte_array_append_uint8 (ota_cmd, 0x9);					/* ota command length */
+	fu_byte_array_append_uint8 (ota_cmd, FU_PXI_DEVICE_CMD_FW_OBJECT_CREATE);	/* ota command */
+	fu_byte_array_append_uint32 (ota_cmd, fu_chunk_get_address (chk), G_LITTLE_ENDIAN);
+	fu_byte_array_append_uint32 (ota_cmd, fu_chunk_get_data_sz (chk), G_LITTLE_ENDIAN);
+
+	/* increase the serial number */
+	self->sn++;
+
+	/* get pixart wireless module for ota object create command */
+	if (!fu_pxi_composite_receiver_cmd (FU_PXI_DEVICE_CMD_FW_OBJECT_CREATE,
+							self->sn,
+							self->model.target,
+							receiver_cmd,
+							ota_cmd, error))
+		return FALSE;
+
+	/* delay for wireless module device get command response*/
+	g_usleep (50 * 1000);
+
+	if (!fu_pxi_wireless_device_set_feature (FU_DEVICE (parent), receiver_cmd->data, receiver_cmd->len, error))
+		return FALSE;
+
+	/* success */
+	return TRUE;
```

Comment:
seems these lines can be combined to just `return fu_pxi...`

Reply:
fixed it.